### ISSUE TITLE
mock_sqs: Unable to retrieve multiple messages in a single request

### ISSUE THREAD

**Issue by Haklim733** [spulec/moto#3006]
I am trying to receive multiple messages per request but I only see one. If I run the code without mocking, I see both messages sent below in the code. I tried introducing time.sleep(3) in between sending messages and receiving the message but I end up with the same results.

**Versions**
boto==2.49.0
boto3==1.13.1
botocore==1.16.1
moto==1.3.14

**Code**
```
@mock_sqs
def test_create_fifo_queue(aws_credentials):
    session = boto3.Session(profile_name='testing')
    sqs = session.client("sqs", region_name="us-east-1")
    today = datetime.today().strftime('%Y-%m-%d')
    queue_name = f"{os.environ['AWS_RETRY_Q']}.fifo"  

    sqs.create_queue(QueueName=queue_name,
                     Attributes={"FifoQueue": "true"})

    response = sqs.get_queue_url(QueueName=queue_name)
    queue_url = response['QueueUrl']
    print(queue_url)

    sqs.send_message(QueueUrl=queue_url,
                    MessageBody=json.dumps(['A']),
                    MessageAttributes={
                        'Date': {'DataType': 'String',
                                 'StringValue': today},
                        'QueryType': {'DataType': 'String',
                                   'StringValue': "RETRY"}
                    },
                     MessageGroupId='test-queue',
                     MessageDeduplicationId=f'{today}-A')

    sqs.send_message(QueueUrl=queue_url,
                    MessageBody=json.dumps(['B']),
                    MessageAttributes={
                        'Date': {'DataType': 'String',
                                 'StringValue': today},
                        'QueryType': {'DataType': 'String',
                                   'StringValue': "NEW"}
                    },
                     MessageGroupId='test-queue',
                     MessageDeduplicationId=f'{today}-B')

    message_resp = sqs.receive_message(QueueUrl=queue_url,
                                       AttributeNames=['All'],
                                       MessageAttributeNames=['All'],
                                       VisibilityTimeout=20,
                                       MaxNumberOfMessages=10,
                                       WaitTimeSeconds=5
                                       )  #this works on production, receive 2 messages
    print('message1', message_resp['Messages'])
    assert len(message_resp['Messages'])==2
```
**traceback**
AssertionError: assert 1 == 2
E        +  where 1 = len([{'Attributes': {'ApproximateFirstReceiveTimestamp': '1589837266039', 'ApproximateReceiveCount': '1', 'MessageDeduplic...', 'MD5OfBody': '5c24fc1df56d764b550ceae1b9319125', 'MD5OfMessageAttributes': '392230c5ed0aa0660ddc3fd110bdd305', ...}])

test_sqs.py:63: AssertionError

**Comment by bblommers** [comment]
Thanks for raising this @Haklim733. Marked it as a bug, and attached a PR that should fix this.

**Comment by bblommers** [comment]
This is now fixed in Moto >= 1.3.15.dev874
